### PR TITLE
Conformance GOV: translate and adjust

### LINKED ISSUE #4468: Conformance GOV: Translate and Adjust

### PR DESCRIPTION
**Pull request by aniketd**
# Description

Resolves #4468 

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs [hunk 1]
```diff
@@ -31,159 +31,59 @@ import Cardano.Ledger.Conway.Governance (
...
-    agdaEnv <-
-      impAnn "Translating the environment" . expectRight . runSpecTransM modifiedCtx $ toSpecRep env
-    logEntry $ "agdaEnv:\n" <> showExpr agdaEnv
-    agdaSt <- impAnn "Translating the state" . expectRight . runSpecTransM modifiedCtx $ toSpecRep st
-    logEntry $ "agdaSt:\n" <> showExpr agdaSt
-    agdaSig <- impAnn "Translating the signal" . expectRight . runSpecTransM modifiedCtx $ toSpecRep sig
-    logEntry $ "agdaSig:\n" <> showExpr agdaSig
-    pure (agdaEnv, agdaSt, agdaSig)
-
-  testConformance ctx env st sig = property $ do
```
Comment (Lucsanszky) [inline libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs:178]: This was not moved to `ExecSpecRule.Conway.Gov` (unless I'm practically blind), so we are basically removing it. Why?

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Gov.hs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import Test.Cardano.Ledger.Conway.Arbitrary ()
+import Test.Cardano.Ledger.Conformance.ExecSpecRule.Conway.Base ()
+import Data.Bifunctor (first)
+
+instance 
+  Inject 
+    (TxId StandardCrypto, Proposals Conway, EnactState Conway) 
+    (EnactState Conway) 
+  where
+  inject (_, _, x) = x
```
Comment (Lucsanszky) [inline libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Gov.hs]: I don't quite see how this is equivalent to the `Inject ConwayGovTransContext` instances from earlier. Or did something change which necessitated this? Is this related to the fact that `testConformance` was removed (see my other comment)?

**Review by Lucsanszky — COMMENTED** [review]
A few things were not quite clear to me, so I added some comments. I'll have a look again tomorrow with a fresh mind.

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs (hunk 1 above)
Comment (aniketd) [reply libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs:178]: This is just overriding the default implementation, and we don't need to do that. I think that this overriding was a left-over from an experiment.

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Gov.hs (hunk 2 above)
Comment (aniketd) [reply libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Gov.hs]: The `ConwayGovTransContext` was introduced in order to provide the `EnactState` to only the Agda side of things, but the same thing can just be achieved with a 3-tuple, which makes for less code overall.

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs [hunk 3]
```diff
@@ -675,35 +673,6 @@ instance
...
-    props <- askCtx @(Proposals era)
-    enactState <- askCtx @(EnactState era)
-    let
-      protVer = gePParams ^. ppProtocolVersionL
-      -- TODO express this in agdaConstraints rather than doing it during
-      -- translation
-      enactStatePV =
-        enactState
-          & ensProtVerL .~ protVer
-          & ensPrevGovActionIdsL .~ toPrevGovActionIds (props ^. pRootsL)
```
Comment (aniketd) [inline libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs:694]: @Lucsanszky you must have noticed that this has been removed and added only to the general `translateInputs` function, while earlier, overriding the `ensPrevGovActionIds` was being done twice: once here and once in `translateInputs`, which was superfluous.

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Gov.hs (hunk 2 above)
Comment (Lucsanszky) [reply libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Gov.hs]: Nice!

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs (hunk 3 above)
Comment (Lucsanszky) [reply libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs:694]: Good catch!

**Review by Lucsanszky — APPROVED** [review]
Thanks for the clarifications @aniketd, LGTM. :+1:

**Comment by teodanciu** [comment]
Force-pushed to rebase and solve conflicts

**Pull request merged**